### ISSUE TITLE
java.lang.NullPointerException: Cannot invoke "java.lang.Enum.name()" because "value" is null

### ISSUE THREAD

**Issue by kevin70** [avaje/avaje-jsonb#77]
Enumeration null value cannot be handled correctly.

stacktrace:
```
java.lang.NullPointerException: Cannot invoke "java.lang.Enum.name()" because "value" is null
io.avaje.jsonb.JsonException: java.lang.NullPointerException: Cannot invoke "java.lang.Enum.name()" because "value" is null
	at app//io.avaje.jsonb.core.DJsonType.toJson(DJsonType.java:82)
	at app//io.avaje.jsonb.core.DJsonType.toJson(DJsonType.java:54)
	at app//io.avaje.jsonb.core.DJsonb.toJson(DJsonb.java:46)
	at app//cool.houge.wujing.JsonbTests.f(JsonbTests.java:56)
	at java.base@17.0.6/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

test source:
```
import io.avaje.jsonb.Json;
import io.avaje.jsonb.Jsonb;
import org.junit.jupiter.api.Test;

/**
 * @author ZY (kzou227@qq.com)
 */
public class JsonbTests {

    @Json
    public static class Staff {

        private String name;
        private StaffStatus status;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public StaffStatus getStatus() {
            return status;
        }

        public void setStatus(StaffStatus status) {
            this.status = status;
        }
    }

    public enum StaffStatus {
        NORMAL(1),
        ;
        public final int code;

        StaffStatus(int code) {
            this.code = code;
        }

        @Json.Value
        public int getCode() {
            return code;
        }
    }

    @Test
    void f() {
        var jsonb = Jsonb.builder().build();
        var bean = new Staff();
        bean.setName("ZY");
        // bean.setStatus(...);
        var str = jsonb.toJson(bean);
        System.out.println(str);
    }

    @Test
    void s() {
        var jsonb = Jsonb.builder().build();
        var bean = new Staff();
        bean.setName("ZY");
        bean.setStatus(StaffStatus.NORMAL);

        var str = jsonb.toJson(bean);
        System.out.println(str);
    }
}
```
Running the `f()` method with junit will get the error stack.

**Comment by rbygrave** [comment]
The fix for this is included in 1.4-RC2 which has just been released to maven central.

**Comment by kevin70** [comment]
Thank you, the bug in the new version has been fixed.
